app/services/zpid_collector.rb

```diff
@@ -0,0 +1,42 @@
+class ZpidCollector
+  def self.fill_missing_zpids
+    total_houses = 0
+    updated_count = 0
+
+    House.includes(:house_metadatum).where(house_metadatum: { zpid: nil }).find_each do |house|
+      begin
+        new(house).get_zpid
+        updated_count += 1
+      rescue ZpidCollectorError => e
+        Rails.logger.error e
+      ensure
+        total_houses += 1
+      end
+    end
+
+    ZpidCollectorResult.new(updated_count: updated_count, total_houses: total_houses)
+  end
+
+  def initialize(house)
+    @house = house
+  end
+
+  def get_zpid
```

Comment:
At first, I felt like this might be a good name, but reading this over I'm not necessarily tooo crazy about it. I'm open to suggestions!

Reply:
`get_zpid` seems reasonable to me